include/deal.II/grid/tria_objects_orientations.h

```diff
@@ -0,0 +1,317 @@
+     * = `true` case as flip-rotate is equal to the identity operation. This
+     * choice ensures that the integer value of the raw orientation is in $[0,
+     * 5]$.
+     */
+    class TriaObjectsOrientations
+    {
+    public:
+      /**
+       * Constructor.
+       */
+      TriaObjectsOrientations();
+
+      /**
+       * Constructor. Sets up objects in the default orientation (orientation
+       * = `true`).
+       */
+      TriaObjectsOrientations(const unsigned int n_objects);
+
+      /**
+       * Return number of geometric objects stored by this class.
+       */
+      unsigned int
+      n_objects() const;
+
+      /**
+       * Reset the object to a default state.
+       */
+      void
+      reinit(const unsigned int n_objects);
+
+      /**
+       * Change the number of stored objects. New objects are constructed in
+       * the default orientation (true, false, false).
+       */
+      void
+      resize(const unsigned int n_objects);
+
+      /**
+       * Return the size of objects of this kind.
+       */
+      std::size_t
+      memory_consumption() const;
+
+      /**
+       * Get the raw orientation of the object, as described in the class
+       * documentation.
+       */
+      unsigned char
+      get_raw_orientation(const unsigned int object) const;
+
+      /**
+       * Get the orientation bit of the object.
+       */
+      bool
+      get_orientation(const unsigned int object) const;
+
+      /**
+       * Get the rotation bit of the object.
+       */
+      bool
+      get_rotation(const unsigned int object) const;
+
+      /**
+       * Get the flip bit of the object.
+       */
+      bool
+      get_flip(const unsigned int object) const;
+
+      /**
+       * Set the raw orientation of the object, as described in the class
+       * documentation.
+       */
+      void
+      set_raw_orientation(const unsigned int object, const unsigned char value);
+
+      /**
+       * Set the orientation bit of the object.
+       */
+      void
+      set_orientation(const unsigned int object, const bool value);
+
+      /**
+       * Set the rotate bit of the object.
+       */
+      void
+      set_rotation(const unsigned int object, const bool value);
+
+      /**
+       * Set the flip bit of the object.
+       */
+      void
+      set_flip(const unsigned int object, const bool value);
+
+      /**
+       * Read or write the data of this object to or from a stream for the
+       * purpose of serialization using the [BOOST serialization
+       * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html).
+       */
+      template <class Archive>
+      void
+      serialize(Archive &ar, const unsigned int version);
+
+    private:
+      /**
+       * Number of objects.
+       */
+      unsigned int n_stored_objects;
+
+      /**
+       * Flags.
+       */
+      std::vector<unsigned char> flags;
```

Comment:
It's a shame we can't use `std::bitset` here instead of `unsigned char` -- but (I tried) `std::bitset<3>` still occupies 8 bytes :-(

Reply:
I tried the same thing but there is a one-word minimum for performance reasons for `std::bitset`. The only way to get sub-byte size objects is with C-style bitfields.